### DISCUSSION TITLE
TabItem content not showing?

### DISCUSSION THREAD

**Discussion by CosminCAS123** [AvaloniaUI/Avalonia#12867] [Q&A]
I created a tabcontrol with 2 tab items for my email app ( inbox and sent).How do I add content to them?
Obviously when changing tabs i ll display different user controls.I checked the **content** property of TabItem with a random button, and it added the button the way i was not expecting:
![image](https://github.com/AvaloniaUI/Avalonia/assets/79158479/6b830ed9-9b7d-4e7f-981d-fff6c46e033d)
Do I have to manually add a ControlContent to the window and change it each time the user switches between tabs?I thought the content property of tabitem will just do that for me.

**Comment by timunie** [comment]
Show us your XAML and ViewModel

**Reply by CosminCAS123** [reply]
i ve just created the window , no viewmodel for now but i ll add one when i understand how the content can be added. xaml is just a tab control with 2 tabitems .

**Reply by timunie** [reply]
normally I'd do it via `DataTemplate` (ItemTemplate and ContentTemplate). Maybe this gives you a good starting point: https://github.com/AvaloniaUI/Avalonia.Samples#%EF%B8%8F-datatemplate-samples
